### ISSUE TITLE
FFRK-LabMem stops working when updating.

### ISSUE THREAD

**Issue by OnSilverwings** [HughJeffner/FFRK-LabMem#103]
I had this happen on the previous update but I wasn't able to document it.

When you press alt+u to update, Windows appears to see it as the software not working and generates  a report (had issues with my pc so I check this often.

The app still updates and works although it says at first it can't cant adb and I have to close everything and start over but then it works fine. I dm'd you in discord a screen capture of what Windows generates the moment LabMem closes down to update itself.

**Comment by HughJeffner** [comment]
Does this happen after it downloads the installer then says "Starting installer and exiting"?

I've seen this happen if you update when adb not connected but I thought I fixed it.

**Comment by OnSilverwings** [comment]
I believe this happens at the moment the app closes to update. You get no notification this happens it's only by checking the report that you see something happened but the update works as normal.

**Comment by HughJeffner** [comment]
Okay, that makes sense.  I'll figure out a more graceful way to close the app.

**Comment by OnSilverwings** [comment]
It's your call. I'm just reporting what I find and let you as the author decide whether it's something that needs fixing. I'm not an expert to know if this indeed needs fixing or it's just Window's picky way of how it regards a clean exit. It's certainly not affecting PC performance and it's only something you'd notice if you're checking this one screen for reliability.
I thought I'd check the event viewer and this is the first error log.

Log Name:      Application
Source:        .NET Runtime
Date:          11/12/2021 11:29:51
Event ID:      1026
Task Category: None
Level:         Error
Keywords:      Classic
User:          N/A
Computer:      Silverwings
Description:
Application: FFRK-LabMem.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: System.ObjectDisposedException
   at System.Threading.CancellationTokenSource.ThrowObjectDisposedException()
   at Titanium.Web.Proxy.ProxyServer.Stop()
   at Titanium.Web.Proxy.ProxyServer.Dispose(Boolean)
   at Titanium.Web.Proxy.ProxyServer.Finalize()


Event Xml:
<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
  <System>
    <Provider Name=".NET Runtime" />
    <EventID Qualifiers="0">1026</EventID>
    <Version>0</Version>
    <Level>2</Level>
    <Task>0</Task>
    <Opcode>0</Opcode>
    <Keywords>0x80000000000000</Keywords>
    <TimeCreated SystemTime="2021-12-11T10:29:51.9653016Z" />
    <EventRecordID>27643</EventRecordID>
    <Correlation />
    <Execution ProcessID="0" ThreadID="0" />
    <Channel>Application</Channel>
    <Computer>Silverwings</Computer>
    <Security />
  </System>
  <EventData>
    <Data>Application: FFRK-LabMem.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: System.ObjectDisposedException
   at System.Threading.CancellationTokenSource.ThrowObjectDisposedException()
   at Titanium.Web.Proxy.ProxyServer.Stop()
   at Titanium.Web.Proxy.ProxyServer.Dispose(Boolean)
   at Titanium.Web.Proxy.ProxyServer.Finalize()

</Data>
  </EventData>
</Event>

**Comment by OnSilverwings** [comment]
Second error log.

Log Name:      Application
Source:        Application Error
Date:          11/12/2021 11:29:52
Event ID:      1000
Task Category: (100)
Level:         Error
Keywords:      Classic
User:          N/A
Computer:      Silverwings
Description:
Faulting application name: FFRK-LabMem.exe, version: 6.0.1.0, time stamp: 0x61ae338d
Faulting module name: KERNELBASE.dll, version: 10.0.19041.1387, time stamp: 0xcfba7f8a
Exception code: 0xe0434352
Fault offset: 0x0012b522
Faulting process ID: 0x794
Faulting application start time: 0x01d7ee79e34c5b22
Faulting application path: G:\Documents\FFRK-LabMem\FFRK-LabMem.exe
Faulting module path: C:\WINDOWS\System32\KERNELBASE.dll
Report ID: 70a41dcf-b40a-4ad3-b498-fdf954125867
Faulting package full name: 
Faulting package-relative application ID: 
Event Xml:
<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
  <System>
    <Provider Name="Application Error" />
    <EventID Qualifiers="0">1000</EventID>
    <Version>0</Version>
    <Level>2</Level>
    <Task>100</Task>
    <Opcode>0</Opcode>
    <Keywords>0x80000000000000</Keywords>
    <TimeCreated SystemTime="2021-12-11T10:29:52.1795094Z" />
    <EventRecordID>27644</EventRecordID>
    <Correlation />
    <Execution ProcessID="0" ThreadID="0" />
    <Channel>Application</Channel>
    <Computer>Silverwings</Computer>
    <Security />
  </System>
  <EventData>
    <Data>FFRK-LabMem.exe</Data>
    <Data>6.0.1.0</Data>
    <Data>61ae338d</Data>
    <Data>KERNELBASE.dll</Data>
    <Data>10.0.19041.1387</Data>
    <Data>cfba7f8a</Data>
    <Data>e0434352</Data>
    <Data>0012b522</Data>
    <Data>794</Data>
    <Data>01d7ee79e34c5b22</Data>
    <Data>G:\Documents\FFRK-LabMem\FFRK-LabMem.exe</Data>
    <Data>C:\WINDOWS\System32\KERNELBASE.dll</Data>
    <Data>70a41dcf-b40a-4ad3-b498-fdf954125867</Data>
    <Data>
    </Data>
    <Data>
    </Data>
  </EventData>
</Event>

**Comment by OnSilverwings** [comment]
Ands third error log.

Log Name:      Application
Source:        Microsoft-Windows-RestartManager
Date:          11/12/2021 11:30:24
Event ID:      10006
Task Category: None
Level:         Error
Keywords:      
User:          SILVERWINGS\Silverwings
Computer:      Silverwings
Description:
Application or service 'adb' could not be shut down.
Event Xml:
<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
  <System>
    <Provider Name="Microsoft-Windows-RestartManager" Guid="{0888e5ef-9b98-4695-979d-e92ce4247224}" />
    <EventID>10006</EventID>
    <Version>0</Version>
    <Level>2</Level>
    <Task>0</Task>
    <Opcode>0</Opcode>
    <Keywords>0x8000000000000000</Keywords>
    <TimeCreated SystemTime="2021-12-11T10:30:24.6580359Z" />
    <EventRecordID>27649</EventRecordID>
    <Correlation />
    <Execution ProcessID="11208" ThreadID="13632" />
    <Channel>Application</Channel>
    <Computer>Silverwings</Computer>
    <Security UserID="S-1-5-21-2429690891-3724908563-2600303247-1001" />
  </System>
  <UserData>
    <RmApplicationEvent xmlns="http://www.microsoft.com/2005/08/Windows/Reliability/RestartManager/">
      <RmSessionId>1</RmSessionId>
      <FullPath>G:\Documents\FFRK-LabMem\adb.exe</FullPath>
      <DisplayName>adb</DisplayName>
      <AppVersion>0</AppVersion>
      <AppType>5</AppType>
      <TSSessionId>1</TSSessionId>
      <Status>17</Status>
      <Pid>4404</Pid>
      <nFiles>3</nFiles>
      <Files>
        <File>G:\Documents\FFRK-LabMem\adb.exe</File>
        <File>G:\Documents\FFRK-LabMem\AdbWinApi.dll</File>
        <File>G:\Documents\FFRK-LabMem\AdbWinUsbApi.dll</File>
      </Files>
    </RmApplicationEvent>
  </UserData>
</Event>

**Comment by HughJeffner** [comment]
I did get it to crash a few times when testing, but it was random. 

I added improvements in 6.2 to hopefully address this but note the upgrade from 6.1 -> 6.2 will still have the issue.

I'll leave this open in case you want to test it after the release

**Comment by OnSilverwings** [comment]
I updated to 6.2 and there is no error report this time, however, the app tracker gave a large bunch of text about an unhandled exception error then closed so wasn't able to copy/paste.

I opened the app again and didn't get an error at first but then when I pressed alt-U there was an unhandled exception bunch of text which I wasn't able to copy/paste because it closed I think as the app updated.

So kind of fixed, but not sure what the unhandled exceptions were about.
